cpp/unreal_plugins/Vehicle/Source/Vehicle/VehiclePawn.cpp

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "Vehicle/VehiclePawn.h"
+
+#include <string>
+#include <vector>
+
+#include <Animation/AnimInstance.h>
+#include <Camera/CameraComponent.h>
+#include <Components/BoxComponent.h>
+#include <Components/InputComponent.h>
+#include <Components/SkeletalMeshComponent.h>
+#include <Engine/CollisionProfile.h>
+#include <UObject/ConstructorHelpers.h>
+
+#include "CoreUtils/Assert.h"
+#include "CoreUtils/Config.h"
+#include "CoreUtils/Log.h"
+#include "CoreUtils/Unreal.h"
+#include "Vehicle/VehicleMovementComponent.h"
+
+AVehiclePawn::AVehiclePawn(const FObjectInitializer& object_initializer) : APawn(object_initializer)
+{
+    SP_LOG_CURRENT_FUNCTION();
+
+    if (!Config::s_initialized_) {
+        return;
+    }
+
+    ConstructorHelpers::FObjectFinder<USkeletalMesh> skeletal_mesh(*Unreal::toFString(Config::get<std::string>("VEHICLE.VEHICLE_PAWN.SKELETAL_MESH")));
+    SP_ASSERT(skeletal_mesh.Succeeded());
+
+    ConstructorHelpers::FClassFinder<UAnimInstance> anim_instance(*Unreal::toFString(Config::get<std::string>("VEHICLE.VEHICLE_PAWN.ANIM_INSTANCE")));
+    SP_ASSERT(anim_instance.Succeeded());
+
+    skeletal_mesh_component_ = CreateDefaultSubobject<USkeletalMeshComponent>(TEXT("AVehiclePawn::skeletal_mesh_component_"));
+    SP_ASSERT(skeletal_mesh_component_);
+    skeletal_mesh_component_->SetSkeletalMesh(skeletal_mesh.Object);
+    skeletal_mesh_component_->SetAnimClass(anim_instance.Class);
+    skeletal_mesh_component_->SetCollisionProfileName(UCollisionProfile::Vehicle_ProfileName);
+    skeletal_mesh_component_->BodyInstance.bSimulatePhysics = true;
+    skeletal_mesh_component_->BodyInstance.bNotifyRigidBodyCollision = true;
+    skeletal_mesh_component_->BodyInstance.bUseCCD = true;
```

Comment:
```
    skeletal_mesh_component_->BodyInstance.bUseCCD = true;
    skeletal_mesh_component_->bBlendPhysics = true;
    skeletal_mesh_component_->SetGenerateOverlapEvents(true);
    skeletal_mesh_component_->SetCanEverAffectNavigation(false);
```

I don't think any of this is necessary. Do we know why these are here?

Reply:
Chaos tutorial uses AWheeledVehiclePawn (https://github.com/EpicGames/UnrealEngine/blob/5.2.0-release/Engine/Plugins/Experimental/ChaosVehiclesPlugin/Source/ChaosVehicles/Private/WheeledVehiclePawn.cpp). The parameters here (except `bBlendPhysics`) are added to mimic that implementation.

Reply:
Ok, in that case, can you put this in a comment?

Reply:
It's even ok to leave `SetIsReplicated` in the code if you prefer, or maybe leave it in our code commented out, as long as there are clear comments saying that our policy in `AVehiclePawn` is to mimic the implementation of `AWheeledVehiclePawn` as closely as possible, and `SetIsReplicated` is in our code for that reason.